core/src/main/java/bio/terra/pearl/core/service/publishing/PublishingUtils.java

```diff
@@ -57,4 +57,12 @@ public static <T extends BaseEntity & Versioned, D extends BaseVersionedJdbiDao<
             service.assignPublishedVersion(entity.getId());
         }
     }
+
+    public static void setPropertyEnumSafe(Object object, String propertyName, Object newValue) throws Exception {
+        if(object.getClass().getDeclaredField(propertyName).getType().isEnum()) {
+            PropertyUtils.setProperty(object, propertyName, Enum.valueOf((Class<Enum>) PropertyUtils.getPropertyType(object, propertyName), newValue.toString()));
```

Comment:
I would pull this out into a helper method `public static <T> stringToEnum(String value, Class<T extends Enum> class)`